content/en/docs/languages/net/instrumentation.md

```diff
@@ -63,320 +279,362 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+If you followed the instructions to [initialize the SDK](#initialize-the-sdk)
+above, you have a `TracerProvider` setup for you already. You can continue with
+[acquiring a tracer](#acquiring-a-tracer).
 
-Note that this sample uses the Console Exporter. If you are exporting to another
-endpoint, you'll have to use a different exporter.
 
 ### Setting up an ActivitySource
 
-Once tracing is initialized, you can configure an
+Anywhere in your application where you write manual tracing code should configure an
 [`ActivitySource`](/docs/concepts/signals/traces/#tracer), which will be how you
 trace operations with [`Activity`](/docs/concepts/signals/traces/#spans)
 elements.
 
-Typically, an `ActivitySource` is instantiated once per app/service that is
-being instrumented, so it's a good idea to instantiate it once in a shared
-location. It is also typically named the same as the Service Name.
+For example:
 
 ```csharp
-using System.Diagnostics;
 
 public static class Telemetry
 {
     //...
 
-    // Name it after the service name for your app.
-    // It can come from a config file, constants file, etc.
-    public static readonly ActivitySource MyActivitySource = new(TelemetryConstants.ServiceName);
+    public static readonly ActivitySource MyActivitySource = new("name", "version");
 
     //...
 }
 ```
+The values of name and version should uniquely identify the Instrumentation Scope, such as the package, module or class name. While the name is required, the version is still recommended despite being optional.
 
-You can instantiate several `ActivitySource`s if that suits your scenario,
-although it is generally sufficient to just have one defined per service.
+It’s generally recommended to define `ActivitySource` once per app/service that is been instrumented, but you can instantiate several `ActivitySource`s if that suits your scenario
 
-### Creating Activities
+In the case of the example app, there are two places where the `ActivitySource` will be instantiated with an appropriate Instrumentation Scope:
 
-To create an [`Activity`](/docs/concepts/signals/traces/#spans), give it a name
-and create it from your
-[`ActivitySource`](/docs/concepts/signals/traces/#tracer).
+First, in the application file `DiceController.cs`:
 
 ```csharp
-using var myActivity = MyActivitySource.StartActivity("SayHello");
-
-// do work that 'myActivity' will now track
-```
-
-### Creating nested Activities
-
-If you have a distinct sub-operation you'd like to track as a part of another
-one, you can create activities to represent the relationship.
+using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
+using System.Net;
 
-```csharp
-public static void ParentOperation()
+public class DiceController : ControllerBase
 {
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+    private ILogger<DiceController> logger;
+    
+    public static readonly ActivitySource MyActivitySource = new("Dice.Server", "1.0.0");
```

Comment:
I'm not sure this sort of approach is what we should be promoting. The ActivitySource should be shared at an assembly level, not at a controller/class level.

Reply:
Hello @martinjt I am not totally clear on what you mean here.
Do you mean the ActivitySource should be initialized in Program.cs and then passed in here?

Reply:
It'd nice to avoid creating new and new ActivitySource per controller/class, and instead use a single one throughout. Unless there is a reason to separate it out, like enabling/disabling telemetry per ActivitySource. For this doc, it is best to use a single ActivitySource.

https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/AspNetCore Shows one way of doing this in an asp.net core application.